routes/artifacts.js

```diff
@@ -190,6 +190,18 @@ api.declare({
         details.bucket  = this.privateBucket.bucket;
       }
       details.prefix    = [taskId, runId, name].join('/');
+      if (input.contentMD5 || input.contentSHA256) {
+        // Remove this check, when contentMD5 and contentSHA256 are required
+        if (!input.contentMD5 || !input.contentSHA256) {
```

Comment:
Basically, I see providing them as using the new API.
So either you use the new API correctly, or you tolerate your legacy request.